### ISSUE TITLE
Doesn't detect multi-line values if they contain a `=` sign

### ISSUE THREAD

**Issue by mgrachev** [dotenv-linter/dotenv-linter#460]
Example:

```env
MULTILINE="{
    'key': 'value='
}"
```

Result:

```
Checking .env
.env:1 QuoteCharacter: The value has quote characters (', ")
.env:2 IncorrectDelimiter: The 'key': 'value key has incorrect delimiter
.env:2 LeadingCharacter: Invalid leading character detected
.env:2 LowercaseKey: The 'key': 'value key should be in uppercase
.env:2 QuoteCharacter: The value has quote characters (', ")
.env:2 UnorderedKey: The 'key': 'value key should go before the MULTILINE key
.env:3 KeyWithoutValue: The }" key should be with a value or have an equal sign
.env:3 LeadingCharacter: Invalid leading character detected

Found 8 problems
```

**Comment by DDtKey** [comment]
The main problem that's there are confusing cases and in addition it conflicts with existing `QuoteChecker`.

We cannot check if the user has forgotten the quotation mark at the end for sure.

For example:
```rust
MULTILINE="{ 
    'key': 'value='
}
SECOND=Value
THIRD="another"
```

But we can do it since it works in env libs/bash, etc. This will be the concern of the user.
